api-inference-community/docker_images/speechbrain/app/pipelines/audio_source_separation.py

```diff
@@ -22,9 +20,36 @@ def __call__(self, inputs: np.array) -> Tuple[np.array, int]:
                 Check `app.validation` if a different sample rate is required
                 or if it depends on the model
         Return:
-            A :obj:`np.array` and a :obj:`int`: The raw waveform as a numpy array, and the sampling rate as an int.
+            est_sources: np.array  
+            
+                The raw waveforms as a numpy array of size T x S, where T is number of time points and S is the number of sources, and the sampling rate as an int.
+
+            fs : int:
+                The sampling frequency for the estimated sources
         """
-        # IMPLEMENT_THIS
-        raise NotImplementedError(
-            "Please implement AudioSourceSeparationPipeline __call__ function"
-        )
+
+
+        fs_model = self.model.hparams.sample_rate
+
+        # we resample to the model's sampling frequency if needed
+        # we assume that the input is at 16kHz
+        mix = torch.from_numpy(inputs)
```

Comment:
The input is not batched, so should we make a fake 1-batch here maybe ?

Also question for the models, does stereo or multi channel audio make sense for some datasets/tasks/models ?

Reply:
Ah ok, yes we should do `.unsqueeze(0)` to introduce a fake batch dimension.. 
Stereo or multichannel might make sense, but the input can be taken as a tensor in that case also, right?